### ISSUE TITLE
Ignore unexpected metric messages

### ISSUE THREAD

**Issue by bmario** [metricq/metricq-db-hta#28]
In conjunction with metricq/metricq-manager#39, we could (and should) ignore unexpected messages, as the manager and the DB are - in theory - on the same page. Thus the configuration should always be okay. And under normal operation, It works fine.

However, there is [this check](https://github.com/metricq/metricq-db-hta/blob/c99c544dd303b5a4c52a0867a15e3e0aa6b6f868/src/async_hta_service.hpp#L234). 

Image the following situation:
- You remove metric M from the DB config
- You restart the DB to apply the change, as you need to
- During the restart, the manager reconfigures the queue, but there is at least one message for metric M in the queue
- During the DB restart, the DB receives this one message for M
- M is no longer in the config for the HTA::Directory, so that will raise an exception, which gets caught and ignored. Fine, but this message also creates an entry in the `mapped_metrics_` entry because, for the wildcard DB config, we need to add mappings on the fly.
- Now, you realize, you didn't mean to remove the metric M
- So you add back the config entry for M
- You send a `reconfigure` RPC to the DB
- The DB performs the reconfiguration, but look again at the linked check from above. The metric M already has an entry in the `mapped_metrics_` member. So its configuration never gets added to the HTA::Directory.
- Now, the messages for metric M will get ignored, as there is no configuration for it in the HTA::Directory

TL;DR Meme:
![The meme police didn't approve this meme, yet](https://user-images.githubusercontent.com/130405/145188716-1a9693d0-6a5a-4373-b1a3-25cf24f60d6e.png)
